hayeon/programmers-1/모의고사.js

```diff
@@ -0,0 +1,35 @@
+function solution(answers) {
+  const result = [];
+
+  const firstPerson = [1, 2, 3, 4, 5];
+  const secondPerson = [2, 1, 2, 3, 2, 4, 2, 5];
+  const thirdPerson = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
+
+  const score = [0, 0, 0];
+
+  for (let i = 0; i < answers.length; i++) {
+    if (answers[i] === firstPerson[i % firstPerson.length]) score[0]++;
+    if (answers[i] === secondPerson[i % secondPerson.length]) score[1]++;
+    if (answers[i] === thirdPerson[i % thirdPerson.length]) score[2]++;
+  }
+
+  const maxScore = Math.max(...score);
+
+  for (let i = 0; i < score.length; i++) {
+    if (maxScore === score[i]) result.push(i + 1);
+  }
+  return result;
+}
```

Comment:
- 하연님 풀이 잘봤습니다 ! 단, 리뷰 드리고 싶은 부분이 많은 부분에서 윤우님 리뷰와  겹치네요 윤우님 리뷰를 꼭 확인해주시길 바랍니다 🙇‍♂️
- [윤우님 PR](https://github.com/Team-Grace/coding-test/pull/35)
- 아래는 개선 코드입니다.

```suggestion
function solution(answers) {
  const result = [];

  // 모든 패턴을 각각의 변수가 아닌 하나의 배열로 관리
  const patterns = [
    [1, 2, 3, 4, 5],
    [2, 1, 2, 3, 2, 4, 2, 5],
    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5],
  ];

  const scores = [0, 0, 0];

  // 각각의 경우를 하드코딩하는게 아닌 반복문으로 개선
  answers.forEach((answer, i) => {
    patterns.forEach((pattern, j) => {
      if (answers[i] === pattern[i % pattern.length]) {
        scores[j]++;
      }
    });
  });

  const maxScore = Math.max(...scores);

  // 이 부분은 그냥 위에서 forEach를 써서 일관성을 위해 리팩토링 했습니다.
  scores.forEach((score, i) => {
    if (maxScore === score) result.push(i + 1);
  });

  return result;
}
```